src/components/nav-bars/language-selector.vue

```diff
@@ -1,21 +1,37 @@
 <script setup lang="ts">
 import { useTranslation } from 'i18next-vue'
+import { storeToRefs } from 'pinia'
 
-const { i18next } = useTranslation()
+import DropdownList from '@/components/common/dropdown-list.vue'
 
-function changeLanguages(event: Event) {
-  if (event.target) {
-    const lang = (event.target as HTMLInputElement).value
-    i18next.changeLanguage(lang)
-  }
+import { useAppStore } from '@/stores/app.store'
+import { statePersistorLangService } from '@/services/state-persistor/state-persistor-lang.service'
+
+const { i18next, t } = useTranslation()
+const { setLang } = useAppStore()
+const { lang } = storeToRefs(useAppStore())
+const availableLanguages = ['fr', 'en', 'de', 'lb'].map(lang => ({
```

Comment:
Don't think it's very important, but may be put in same order as current prod...

Reply:
Good point. + added a UT to check the order.